File: Matrix/Matrix.c
#include <stdio.h>

int main(){
  int matrix1[] = {
  {1,2},
  {3,4},
  {5,6}
};

int matrix2[] = transponieren(matrix1)
  
for(int i = 0; i < sizeof(matrix2)/sizeof(matrix[0]); i++) {
    for(int j = 0; j < sizeof(matrix2[0])/sizeof(matrix[0][0]); j++) {
        printf("%d ", array[i][j]);
    }
    printf("\n");
} 
}

int transponieren(int matrix){
  int ergebnis[]
  
  for(int i = 0; i < sizeof(matrix[0]) / sizeof(matrix[0][0]); i++){
    ergebnis[i] = {}
  }

  int length = sizeof(matrix) / sizeof(matrix[0])
  
  for(int i = 0; i < length; i++){
    for(int j = 0; j < sizeof(matrix[i]) / sizeof(matrix[i][0]); j++){
      ergebnis[j][i] = matrix[i][j]
    }
  }
  return ergebnis
}

File: Matrix/matrix.js
class Matrix {
    constructor(array){
        this.matrix = array
    }

    transpone(matrixA){

    }

    static mul(matrixA, matrixB){
        let aM = matrixA.length
        let aN = matrixA[0].length

        let bM = matrixB.length
        let bN = matrixB[0].length

        if(aN != bM){
            return undefined
        }

        let matrixC = []

        for(let cM = 0; cM < aM; cM++){
            matrixC[cM] = []
            for(let cN = 0; cN < bN; cN++){
                let result = 0

                for(let i = 0; i < aM; i++){
                    result += matrixA[cN][i] * matrixB[i][cN]
                }

                matrixC[cM][cN] = result
            }
        }

        return matrixC
    }

    get trans(){
        return this.transpone(this.matrix)
    }
}

let matrix1 = [
    [3,2,1],
    [1,0,2]
]

let matrix2 = [
    [1,2],
    [0,1],
    [4,0]
]

let matrix3 = Matrix.mul(matrix1, matrix2)
console.log(matrix3)